src/ngraph/runtime/cpu/cpu_backend.cpp

```diff
@@ -91,17 +91,23 @@ shared_ptr<runtime::Executable>
                                        bool performance_counters_enabled)
 {
     shared_ptr<runtime::Executable> rc;
-    auto it = m_exec_map.find(func);
-    if (it != m_exec_map.end())
+    // we will protect the access to map (m_exec_map) across multiple threads by creating a lock_gaurd
+    // m_exec_map_mutex will be released once the object `guard` goes out of scope
     {
-        rc = it->second;
+        std::lock_guard<std::mutex> guard(m_exec_map_mutex);
+        auto it = m_exec_map.find(func);
+        if (it != m_exec_map.end())
+        {
+            rc = it->second;
+        }
+        return rc;
```

Comment:
That return should be inside the if, no ?

Reply:
ah right!. thanks for catch it.